hubclient/client.go

```diff
@@ -409,6 +422,16 @@ func (c *Client) ForEachPage(link string, listOptions *hubapi.GetListOptions, li
 	return err
 }
 
+// Sets header values to use for any subsequent requests by this client
```

Comment:
If https://golang.org/pkg/net/http/#Header is used, then you can just create Header() method that returns reference to that object, and all `Get`, `Set`, `Add` and `Clear` will become kind of automatic.

Reply:
Each request is created inline and not directly accessible when calling client methods so that would need to be called.... in the middle of a method right now I think.  Though I may be mistaken.

Reply:
Yes, and each request should probably just copy headers into request and populate whatever it needs additionally. But pre-setting these values is exposed through public functions, that what I was talking about.